pipeline/backend/kubernetes/service_test.go

```diff
@@ -24,24 +24,24 @@ import (
 )
 
 func TestServiceName(t *testing.T) {
-	name, err := serviceName(&types.Step{Name: "database"})
+	name, err := serviceName(&types.Step{Name: "database", UUID: "01he8bebctabr3kgk0qj36d2me-0"})
 	assert.NoError(t, err)
-	assert.Equal(t, "database", name)
+	assert.Equal(t, "database-01he8bebctabr3kgk0qj36d2me-0", name)
 
-	name, err = serviceName(&types.Step{Name: "wp-01he8bebctabr3kgk0qj36d2me-0-services-0.woodpecker-runtime.svc.cluster.local"})
+	name, err = serviceName(&types.Step{Name: "wp-01he8bebctabr3kgk0qj36d2me-0-services-0.woodpecker-runtime.svc.cluster.local", UUID: "01he8bebctabr3kgk0qj36d2me-0"})
 	assert.NoError(t, err)
-	assert.Equal(t, "wp-01he8bebctabr3kgk0qj36d2me-0-services-0.woodpecker-runtime.svc.cluster.local", name)
+	assert.Equal(t, "wp-01he8bebctabr3kgk0qj36d2me-0-services-0.woodpecker-runtime.svc.cluster.local-01he8bebctabr3kgk0qj36d2me-0", name)
 
-	name, err = serviceName(&types.Step{Name: "awesome_service"})
+	name, err = serviceName(&types.Step{Name: "awesome_service", UUID: "01he8bebctabr3kgk0qj36d2me-0"})
 	assert.NoError(t, err)
-	assert.Equal(t, "awesome-service", name)
+	assert.Equal(t, "awesome-service-01he8bebctabr3kgk0qj36d2me-0", name)
 }
 
 func TestService(t *testing.T) {
 	expected := `
 	{
 	  "metadata": {
-	    "name": "bar",
+	    "name": "bar-01he8bebctabr3kgk0qj36d2me-0",
 	    "namespace": "foo",
 	    "creationTimestamp": null
 	  },
```

Comment:
i guess we should add "hostnames" key to spec ...!

Reply:
https://kubernetes.io/docs/tasks/network/customize-hosts-file-for-pods/#adding-additional-entries-with-hostaliases

Reply:
The hostname remains the same, just the `step.Name` to use in the pipeline. 

Name with UUID is just for not conflict with others k8s resources

Reply:
yes if I'm correct the hostname was never set as it should be ...

.. I have to setup an k3s again to test services with cube ...

Reply:
the question is if they do not work, if you could also add the fix in here or create a new pull

if it works now just fine already - I'm happy :)

Reply:
In my tests, it already works as expected. 

Here I enter two different pod steps running they set `mydb` for each pipeline. 

![image](https://github.com/woodpecker-ci/woodpecker/assets/22245125/887be3dc-7fe2-4c0e-bb72-97c80cb8b4c0)

Reply:
> if I'm correct the hostname was never set as it should be

It worked before changes mentioned in #3178. Service name was unique, but hostalias (/etc/hosts) contained just name and IP, like `10.10.10.10 postgres`.